core/lib/eth_client/src/clients/http/query.rs

```diff
@@ -266,6 +266,20 @@ where
         Ok(logs)
     }
 
+    async fn logs_extended(
+        &self,
+        filter: &web3::Filter,
+    ) -> EnrichedClientResult<Vec<zksync_types::api::Log>> {
```

Comment:
I personally agree that this is the easiest choice, but I recall I had to refactor my types with some pain to ensure that certain methods are available only for types that do support it (e.g. I added L2-like base fee history only for `ForWeb3Network<Net = L2>`). 

But if you decide to keep it this way, it may make sense to add a comment on what will happen if someone tries to use the method with l1 provider

Reply:
I have reworked eth watcher client